client/components/settings/local_settings_modal_controller.js

```diff
@@ -0,0 +1,33 @@
+class LocalSettingsModalController {
+  constructor (localSettings) {
+    Object.assign(this, { localSettings })
+    this.settings = {}
+  }
+
+  $onInit () {
+    Object.assign(this.settings, this.localSettings.get())
+  }
+
+  loadSettings () {
+    const fromService = this.localSettings.get()
+    const theKeys = Object.keys(fromService)
+    this.settingsKeys = theKeys
+    this.settingsCopy = fromService
+    this.forDOM = []
+    this.settingsKeys.forEach(key => {
+      this.forDOM.push({
+        keyName: key,
```

Comment:
Why not just key? that way you can also do `{ key, data: this.settingsCopy[key] }`. Why are you making so many copies of the settings? Why not use it directly as is?

Reply:
what do you mean "directly as is"?